src/redux/books/books.js

```diff
@@ -1,46 +1,42 @@
-import { addBookApi, getBookApi, removeBookApi } from '../../Api/booksApi';
+import { apiAddBook, apiGetBooks, apiRemoveBook } from '../../modules/api';
 
-const ADD_BOOK = 'bookstore/books/ADD_BOOK';
-const REMOVE_BOOK = 'bookstore/books/REMOVE_BOOK';
-const GET_BOOK = 'bookstore/books/GET_BOOK';
-
-const initialState = [];
+const ADD = 'bookstore/Book/ADD';
+const GET = 'bookstore/Book/GET';
+const REMOVE = 'bookstore/Book/REMOVE';
 
 export const addBook = (payload) => async (dispatch) => {
-  const book = { ...payload, item_id: payload.id };
-  await addBookApi(book);
-  dispatch({ type: ADD_BOOK, book });
-};
-
-export const removeBook = (id) => async (dispatch) => {
-  await removeBookApi(id);
-  dispatch({ type: REMOVE_BOOK, id });
+  const book = { ...payload, item_id: payload.id, category: 'category' };
+  await apiAddBook(book);
+  dispatch({ type: ADD, book });
 };
 
-export const getBook = () => async (dispatch) => {
-  const data = await getBookApi();
-  const books = Object.keys(data).map((id) => {
-    const book = data[id][0];
-    book.id = id;
+export const getBooks = () => async (dispatch) => {
+  const data = await apiGetBooks();
+  const books = Object.keys(data).map((key) => {
+    const book = data[key][0];
+    book.id = key;
     return book;
   });
-  dispatch({ type: GET_BOOK, books });
+  dispatch({ type: GET, books });
+};
+
+export const removeBook = (id) => async (dispatch) => {
+  await apiRemoveBook(id);
+  dispatch({ type: REMOVE, id });
 };
 
-const reducer = (state = initialState, action) => {
+export default function reducer(state = [], action = {}) {
   switch (action.type) {
-    case ADD_BOOK:
+    case ADD:
       return [...state, action.book];
 
-    case REMOVE_BOOK:
-      return [...state].filter((book) => book.id !== action.id);
-
-    case GET_BOOK:
+    case GET:
       return action.books;
 
+    case REMOVE:
+      return [...state].filter((book) => book.id !== action.id);
```

Comment:
- [ ] just as I also mentioned in my previous review, kindly look up the redux toolkit to know how to use it with redux

Reply:
@VICTOR-CHUKWU  I submitted the wrong link for that PR this is actually the first reveiw,

I tested it again, kind sir, I'm also not clear on the problems  on coorecton 2 and 3 pls can u clarrify

Reply:
send me a message on slack so i can explain since you are not clear on it yet

Reply:
hi  @VICTOR-CHUKWU 

not on slack is zoom chat ok

Reply:
send a zoom message @okenwa pablo